lib/setup.js

```diff
@@ -131,19 +153,29 @@ function isSemver(s) {
  * @returns {Promise<string>} The directory SAM CLI is installed in.
  */
 // TODO: Support more platforms
-async function installUsingNativeInstaller(version) {
+async function installUsingNativeInstaller(inputVersion) {
   if (os.platform() !== "linux" || os.arch() !== "x64") {
     core.setFailed("Only Linux x86-64 is supported with use-installer: true");
     return "";
   }
 
   // Must be full semantic version; downloads version directly from GitHub
-  if (version && !isSemver(version)) {
+  if (inputVersion && !isSemver(inputVersion)) {
     core.setFailed("Version must be in the format x.y.z");
     return "";
   }
 
-  // TODO: If not set, check latest version and return from cache if exists
+  let version = inputVersion;
+  if (!version) {
+    const latestVersion = await getLatestReleaseTag("aws/aws-sam-cli");
+    // Must be full semantic version; downloads version directly from GitHub
+    if (latestVersion && !isSemver(latestVersion)) {
+      core.setFailed("Latest version found must be in the format x.y.z");
+      return "";
```

Comment:
If for some reason the tag is messed up, it’ll break the installation. Could instead let it download the latest version without caching, and log the error.

Reply:
Good point. Will have a subsequent PR to fix it.

Reply:
Actually I think one of the unit test covered this case and it will allow install the latest version directly.

Reply:
Ah, yeah that one works when there's a network error. But if the call succeeds and the version doesn't pass `isSemver()`, it'll fail. It's probably fine, although it won't work if the pattern changes.

Reply:
make sense. going to push a fix